vdr/doc/resolvers.go

```diff
@@ -0,0 +1,125 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package doc contains DID Document related functionality that only matters to the current node.
+// All functionality here has zero relations to the network.
+package doc
+
+import (
+	"crypto"
+	"fmt"
+	"time"
+
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/vdr/types"
+)
+
+// Resolver implements the DocResolver interface with a types.Store as backend
+type Resolver struct {
```

Comment:
I find the naming of these types a bit odd, partly because we abbreviate `Document` to `Doc` in `DocResolver` which we never somewhere else, and because the type name of the implementation of the interface is shorter and less-specific than the interface. It might be my Java background, but I'd think of one of the options below;

- StoreResolver implements Resolver
- StoreDocResolver implements DocResolver
- StoreDocumentResolver implements DocumentResolver

Reply:
There's more to do in the VDR package to clean stuff up. 
The implementation is missing the `Doc` prefix because of the package name, it's used as `doc.Resolver`.
The biggest problem is that both the `Store` and `Resolver` implement the same interface. Refactoring that is beyond the scope of this PR.